Fix #24 setup process manager with apache airflow

curioswati opened the pull request.

Commit:
add apache-airflow in deps

Commit:
add dag definitions for running scrapy crawls

Commit:
add postgres deps in pipfile

Commit:
rename the airflow direcories

Commit:
add task to create a `datasets` dir

Commit:
update dag definition for crawl

Commit:
use jinja template vars to provide arguments to tasks

Commit:
replace TriggerOperator with ShortCircuitOperator

Commit:
use a single dag for all the tasks

Commit:
add branching to make appropriate dependencies in task execution

Commit:
add time delay between crawls

Commit:
update the way ddo code csv is read to avoid a pylint error

apoorv74 commented:
hey @curioswati did you try running this for a sample ? also, can you share the link to the airflow dashboard, we can use that for reporting logs and other metrics

curioswati commented:
> hey @curioswati did you try running this for a sample ? also, can you share the link to the airflow dashboard, we can use that for reporting logs and other metrics

Yes @apoorv74. The sample run is in progress and the dashboard is live.

Commit:
update airflow extra package deps in Pipfile

Commit:
update dag schedule

gggodhwani commented on scheduler/tasks/crawl_treasuries.py:
```diff
@@ -0,0 +1,81 @@
+
+DEFAULT_ARGS = {
+    'owner': 'airflow',
+    'start_date': dt.datetime(2019, 8, 1),
```
Can we please set it to the first of the month whenever the call happen?

gggodhwani commented on scheduler/tasks/crawl_treasuries.py:
```diff
@@ -0,0 +1,81 @@
+    'owner': 'airflow',
+    'start_date': dt.datetime(2019, 8, 1),
+    'concurrency': 1,
+    'retries': 0
```
Can we explain the reason of keeping the retry count keeping 0?

gggodhwani commented on scheduler/tasks/crawl_treasuries.py:
```diff
@@ -0,0 +1,81 @@
+
+with DAG('crawl_treasuries',
+         default_args=DEFAULT_ARGS,
+         schedule_interval='30 4 * * *',
```
Can we explain the CRON equivalent & timezone?

gggodhwani commented on scheduler/tasks/crawl_treasuries.py:
```diff
@@ -0,0 +1,81 @@
+    )
+
+    EXP_CRAWL_COMMAND = Template("""
+        cd $project_path/scraper && scrapy crawl treasury_expenditures -a start={{ ds_nodash }} -a end={{ ds_nodash }}
```
Can we please link the Airflow docs here to explain Jinja variables if any?

gggodhwani commented on scheduler/tasks/crawl_treasuries.py:
```diff
@@ -0,0 +1,81 @@
+        cd $project_path/scraper && scrapy crawl treasury_expenditures -a start={{ ds_nodash }} -a end={{ ds_nodash }}
+    """)
+
+    CRAWL_EXPENDITURE = BashOperator(
```
Can we make the nomenclature consistent, something like: EXP_CRAWL_TASK

gggodhwani commented on scheduler/tasks/crawl_treasuries.py:
```diff
@@ -0,0 +1,81 @@
+        cd $project_path/scraper && scrapy crawl treasury_receipts -a start={{ ds_nodash }} -a end={{ ds_nodash }}
+    """)
+
+    CRAWL_RECEIPTS = BashOperator(
```
Same nomenclature thing

gggodhwani commented on scraper/scraper/spiders/treasury_base.py:
```diff
@@ -127,6 +132,8 @@ def parse(self, response):
                 }
                 yield self.make_dataset_request(params)
 
+            time.sleep(5)
```
Comment: # add a sleep of 5 seconds after each treasury crawled

gggodhwani reviewed (changes requested):
Requesting some minor changes

Commit:
minor updates

- change the dag start date to dynamically set to the first date of the
month when the dag is called

- change variable names for tasks

Commit:
add more comments

gggodhwani reviewed (approved):
:+1:
